File: analyzers/tests/SonarAnalyzer.Test/TestCases/PropertyGetterWithThrow.cs
namespace Tests.Diagnostics
{
    using System;

    public class PropertyGetterWithThrow
    {
        public int MyProperty
        {
            get
            {
                var x = 5;
                throw new NotSupportedException(); // Compliant
            }
        }
        public int MyProperty2
        {
            get
            {
                throw new NotImplementedException(); // Compliant
            }
        }
        public int MyProperty3
        {
            get
            {
                throw new PlatformNotSupportedException(); // Compliant
            }
        }
        public int MyProperty4
        {
            get
            {
                throw new Exception(); // Noncompliant {{Remove the exception throwing from this property getter, or refactor the property into a method.}}
//              ^^^^^^^^^^^^^^^^^^^^^^
            }
        }
        public int MyProperty5
        {
            get
            {
                return 42;
            }
            set
            {
                throw new Exception(); // Compliant - setters are ignored by this rule
            }
        }
        public int MyProperty6
        {
            get
            {
                throw new InvalidOperationException(); // Compliant
            }
        }
        public int MyProperty7
        {
            get
            {
                throw new ObjectDisposedException(""); // Compliant
            }
        }
        public int MyProperty8
        {
            get
            {
                try
                {
                }
                catch(Exception)
                {
                    throw; // Compliant
                }

                return 0;
            }
        }
        public int this[int i]
        {
            get
            {
                throw new System.Exception(); // Compliant - indexed getters are ignored by this rule
            }
            set
            {
            }
        }
        public int MyProperty9
        {
            get
            {
                throw FactoryMethod(); // Compliant
            }
        }
        private NotSupportedException FactoryMethod()
        {
            return new NotSupportedException();
        }
        public int MyProperty10
        {
            get
            {
                throw FactoryMethod2(); // Noncompliant {{Remove the exception throwing from this property getter, or refactor the property into a method.}}
//              ^^^^^^^^^^^^^^^^^^^^^^^
            }
        }
        private Exception FactoryMethod2()
        {
            return new Exception();
        }
    }
}

File: analyzers/tests/SonarAnalyzer.Test/TestCases/PropertyGetterWithThrow.vb
Imports System
Namespace Tests.Diagnostics
    Public Class PropertyGetterWithThrow
        Public Property Foo1() As Integer
            Get
                Throw New Exception() ' Noncompliant
'               ^^^^^^^^^^^^^^^^^^^^^
            End Get
            Set(ByVal value As Integer)
                ' ... some code ...
            End Set
        End Property
        Public Property Foo2() As Integer
            Get
                Throw New NotImplementedException() ' Compliant
            End Get
            Set(ByVal value As Integer)
                ' ... some code ...
            End Set
        End Property
        Public Property Foo3() As Integer
            Get
                Throw New NotSupportedException() ' Compliant
            End Get
            Set(ByVal value As Integer)
                ' ... some code ...
            End Set
        End Property
        Public Property Foo4() As Integer
            Get
                Throw New PlatformNotSupportedException() ' Compliant
            End Get
            Set(ByVal value As Integer)
                ' ... some code ...
            End Set
        End Property
        Public ReadOnly Property Foo5() As Integer
            Get
                Return 1
            End Get
        End Property
        Public ReadOnly Property Foo6() As Integer
            Get
                Throw New InvalidOperationException() ' Compliant
            End Get
        End Property
        Public ReadOnly Property Foo7() As Integer
            Get
                Throw New ObjectDisposedException("foo") ' Compliant
            End Get
        End Property
        Public Property Item(ByVal index As Long) As Byte ' Indexed properties are ignored
            Get
                Throw New NotImplementedException
            End Get
            Set(ByVal Value As Byte)

            End Set
        End Property
        Public ReadOnly Property Foo9() As Integer
            Get
                Throw FactoryMethod() ' Compliant
            End Get
        End Property
        private Function FactoryMethod() as NotSupportedException
            return new NotSupportedException()
        End Function
        Public ReadOnly Property Foo10() As Integer
            Get
                Throw FactoryMethod2() ' Noncompliant
'               ^^^^^^^^^^^^^^^^^^^^^^
            End Get
        End Property
        private Function FactoryMethod2() as Exception
            return new Exception()
        End Function
    End Class
End Namespace
